### ISSUE TITLE
[bug] minecraft crashes on joining world using latest version of mousewheelie

### ISSUE THREAD

**Issue by mrmrchoice** [Siphalor/mouse-wheelie#16]
Crashes consistently when joining either single-player (when world gen complete) or multiplayer server. If I swap to the previous version there is no crash.

---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 08/06/19 13:32
Description: Unexpected error

java.lang.NoSuchMethodError: net.fabricmc.fabric.impl.network.PacketRegistryImpl.createInitialRegisterPacket(Lnet/fabricmc/fabric/api/network/PacketRegistry;)Ljava/util/Optional;
	at net.minecraft.class_634.handler$onGameJoin$zzc000(class_634.java:2260)
	at net.minecraft.class_634.method_11120(class_634.java:376)
	at net.minecraft.class_2678.method_11567(class_2678.java:76)
	at net.minecraft.class_2678.method_11054(class_2678.java:13)
	at net.minecraft.class_2600.method_11072(class_2600.java:15)
	at net.minecraft.class_1255.method_18859(class_1255.java:135)
	at net.minecraft.class_4093.method_18859(class_4093.java:23)
	at net.minecraft.class_1255.method_16075(class_1255.java:114)
	at net.minecraft.class_1255.method_5383(class_1255.java:99)
	at net.minecraft.class_310.method_1523(class_310.java:915)
	at net.minecraft.class_310.method_1514(class_310.java:411)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at net.minecraft.class_634.handler$onGameJoin$zzc000(class_634.java:2260)
	at net.minecraft.class_634.method_11120(class_634.java:376)
	at net.minecraft.class_2678.method_11567(class_2678.java:76)
	at net.minecraft.class_2678.method_11054(class_2678.java:13)
	at net.minecraft.class_2600.method_11072(class_2600.java:15)
	at net.minecraft.class_1255.method_18859(class_1255.java:135)
	at net.minecraft.class_4093.method_18859(class_4093.java:23)
	at net.minecraft.class_1255.method_16075(class_1255.java:114)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [class_746['joeltaylor'/222, l='MpServer', x=8.50, y=65.00, z=8.50]]
	Chunk stats: MultiplayerChunkCache: 729, 0
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (8,64,8), Chunk: (at 8,4,8 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 0 game time, 0 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Server brand: ~~NULL~~
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:421)
	at net.minecraft.class_310.method_1587(class_310.java:1920)
	at net.minecraft.class_310.method_1514(class_310.java:431)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- System Details --
Details:
	Minecraft Version: 1.14
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_211, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 848193288 bytes (808 MB) / 1316487168 bytes (1255 MB) up to 2863661056 bytes (2731 MB)
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms204m -Xmx3072m
	Fabric Mods: 
		autoconfig1: 1.0.1+mc1.14
		cloth-config: 0.2.4+build.17
		fabric: 0.2.7+build.127
		fabric-api-base: 0.1.0+59147463
		fabric-events-interaction-v0: 0.1.0+59147463
		fabric-events-lifecycle-v0: 0.1.0+59147463
		fabric-keybindings: 0.1.0+93af775a
		fabric-mining-levels-v0: 0.1.0+59147463
		fabric-networking-v0: 0.1.2+200eb5c2
		fabric-resource-loader: 0.1.0+93af775a
		fabric-resource-loader-v0: 0.1.1+59147463
		fabric-tag-extensions-v0: 0.1.0+59147463
		fabricloader: 0.4.8+build.154
		foamfix: 0.12.1
		modmenu: 1.5.3-84
		mousewheelie: 1.2.2+1.14.2
		simplezoom: 1.0.0
		tweed: 2.1.0-beta.7
		voxelmap: 1.9.2
	Launched Version: MultiMC5
	LWJGL: 3.2.1 build 12
	OpenGL: Intel(R) HD Graphics 620 GL version 4.5.0 - Build 25.20.100.6374, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	CPU: 4x Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz

**Comment by Siphalor** [comment]
The latest versions of Mouse Wheelie are built for Minecraft 1.14.2.
Older version like 1.14.0 might not be compatible.
Since this crash is related to an incompatibility with Fabric's API I could maybe build a version for 1.14.0 but I don't want to be fully backwards compatible.

TL;DR: You may want to update to a newer version of Minecraft or use an older version of Mouse Wheelie.

**Comment by mrmrchoice** [comment]
Hi, I have just updated my Fabric to 1.14.2 and also my server to MC 1.14.2 and the new version still crashes and the old version still works

**Comment by Siphalor** [comment]
Please try to update all other mods too (and delete Fabric API if installed seperately)
